internal/terraform/node_resource_plan_instance.go

```diff
@@ -586,12 +597,28 @@ func (n *NodePlannableResourceInstance) importState(ctx EvalContext, addr addrs.
 		},
 		override: n.override,
 	}
-	instanceRefreshState, refreshDiags := riNode.refresh(ctx, states.NotDeposed, importedState)
+	instanceRefreshState, deferred, refreshDiags := riNode.refresh(ctx, states.NotDeposed, importedState)
 	diags = diags.Append(refreshDiags)
 	if diags.HasErrors() {
 		return instanceRefreshState, diags
 	}
 
+	if deferred != nil {
+		var diags tfdiags.Diagnostics
+		diags = diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"Cannot import deferred remote object",
+			fmt.Sprintf(
+				"While attempting to import an existing object to %q, "+
+					"the provider detected that the object was deferred. "+
+					"Only not-deferred objects can be imported; please make sure "+
+					"the provider has all the information needed to plan the change.",
+				n.Addr,
+			),
+		))
+		return instanceRefreshState, diags
```

Comment:
This I think think is left over from before? It's okay to try and import into a deferred object, we'll just perform the import properly at the next plan.

Reply:
Oh yeah, I just removed the other case :D